### PR TITLE
Update installation instructions

### PR DESCRIPTION
**Pull request by caleb-johnson**
The installation instructions were thorough and correct, but they needed a facelift, and they needed the PyPI installation added as an option.

I lightly borrowed from the [Qiskit installation instructions](https://qiskit.org/documentation/getting_started.html) for language and flow.

### CODE REVIEW

File: INSTALL.rst [hunk 1]
```diff
@@ -1,51 +1,114 @@
...
+.. _Option 1:
+
+Option 1: Pip Installation
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Upgrade pip and install the CKT package.
+
+.. code:: sh
+
+    pip install --upgrade pip circuit-knitting-toolbox
```
Comment (garrison) [inline INSTALL.rst:57]: If you're going to go through the effort of upgrading pip, you should really do this _before_ using pip to install ckt.

File: INSTALL.rst [hunk 2]
```diff
@@ -1,51 +1,114 @@
...
+
+.. code:: sh
+    
+    pip install 'circuit-knitting-toolbox[pyscf]'
+
+Users intending to use the automatic cut finding functionality in the CutQC package should install the ``cplex`` option.
+
+.. code:: sh
+    
+    pip install 'circuit-knitting-toolbox[cplex]'
```
Comment (garrison) [inline INSTALL.rst:69]: Perhaps worth mentioning after these that you can pass multiple in one command, like `pip install 'circuit-knitting-toolbox[cplex,pyscf]'`.

But maybe this makes it too wordy.  If you've already considered this, please disregard.

File: INSTALL.rst [hunk 3]
```diff
@@ -1,51 +1,114 @@
...
--  **OPTIONAL** If a user wishes to use the circuit cutting tool to
-   automatically find optimized wire cuts for a circuit too large for
-   the free version of CPLEX, they should acquire a license and install
-   the `full
-   version <https://www.ibm.com/products/ilog-cplex-optimization-studio>`__.
+Option 2: Local Installation
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
--  Enter a Python environment and install the software
+Clone the CKT repository.
```
Comment (garrison) [inline INSTALL.rst:77]:
```suggestion
Clone the circuit-knitting-toolbox repository.
```

File: INSTALL.rst [hunk 4]
```diff
@@ -1,51 +1,114 @@
...
+    
+
+.. _Option 2:
 
--  **OPTIONAL** If a user wishes to use the circuit cutting tool to
-   automatically find optimized wire cuts for a circuit too large for
-   the free version of CPLEX, they should acquire a license and install
-   the `full
-   version <https://www.ibm.com/products/ilog-cplex-optimization-studio>`__.
+Option 2: Local Installation
```
Comment (garrison) [inline INSTALL.rst:74]: Option 1 also qualifies as "local installation"

File: INSTALL.rst [hunk 5]
```diff
@@ -90,8 +153,39 @@ Once the container is running, you should see a message like this:
...
+Jupyter Notebook interface.
 
 The home directory includes a subdirectory named ``persistent-volume``.
 All work you’d like to save should be placed in this directory, as it is
 the only one that will be saved across different container runs.
+
+
+Running some Examples
+^^^^^^^^^^^^^^^^^^^^^
+From inside the ``circuit_knitting_toolbox`` repository, navigate to the tutorials, open
```
Comment (garrison) [inline INSTALL.rst:165]:
```suggestion
From inside the ``circuit-knitting-toolbox`` repository, navigate to the tutorials, open
```

File: INSTALL.rst [hunk 6]
```diff
@@ -90,8 +153,39 @@ Once the container is running, you should see a message like this:
...
+a `Jupyter Notebook <https://jupyter.org/install>`__ instance, and start experimenting with CKT!
+
+.. code::
+    
+    cd docs/<circuit_cutting | entanglement_forging>/tutorials
+    jupyter notebook
```
Comment (garrison) [inline INSTALL.rst:171]: Why not start `jupyter` from the base of the repository -- or at most, one level deep, in `docs`?

File: INSTALL.rst (hunk 2 above)
Comment (caleb-johnson) [reply INSTALL.rst:69]: I think keeping both "Users intending to..." statements, but collapsing the call into one call containing both options is the most concise

File: INSTALL.rst (hunk 4 above)
Comment (caleb-johnson) [reply INSTALL.rst:74]: I thought "local installation" meant local to the repository location (`-e` flag). Not local to the machine. Every install is inherently a local install

Comment (caleb-johnson) [reply INSTALL.rst:74]: I think Option 2 should be called "Install from Source". You're right, local install is not the right call

File: INSTALL.rst (hunk 6 above)
Comment (caleb-johnson) [reply INSTALL.rst:171]: Couldn't really settle on what was best. This section is kind of clunky anyway, but I think we need it to cap everything off.

What snippet would you suggest here?

File: INSTALL.rst (hunk 1 above)
Comment (garrison) [reply INSTALL.rst:57]: Yeah, that works, or you can do it on two separate lines.

A big part of the reason for upgrading pip is to get the smartest possible dependency resolution, which can only then happen once pip is run a second time.

File: INSTALL.rst (hunk 6 above)
Comment (garrison) [reply INSTALL.rst:171]: Maybe just

```
cd docs/
jupyter notebook
```

and then navigate thru the web interface to circuit_cutting or entanglement_forging, then tutorials

File: INSTALL.rst [hunk 7]
```diff
@@ -1,51 +1,114 @@
...
-    pip install --upgrade pip
+    
+Upgrade pip and enter the repository. 
+
+.. code:: sh
+    
+    pip install --upgrade pip && cd circuit-knitting-toolbox
+
+Install CKT from source. Install the notebook dependencies in order to run
+all the visualizations in the tutorial notebooks.
```
[the remark this answers is not in the document]
Comment (caleb-johnson) [reply INSTALL.rst:90]: OK I've reworked this section. Let me know what you think

File: INSTALL.rst [hunk 8]
```diff
@@ -90,8 +155,39 @@ Once the container is running, you should see a message like this:
...
+
+Running some Examples
+^^^^^^^^^^^^^^^^^^^^^
+From inside the ``circuit-knitting-toolbox`` repository, navigate to the tutorials via the `docs/` directory, open
+a `Jupyter Notebook <https://jupyter.org/install>`__ instance, and start experimenting with CKT!
+
+.. code::
+    
+    cd docs/
+    jupyter notebook
```
Comment (garrison) [inline INSTALL.rst:173]: I actually think this whole thing should be a sub-subsection under Option 2.  It isn't relevant to the other options.

File: INSTALL.rst [hunk 9]
```diff
@@ -1,51 +1,116 @@
...
+.. code:: sh
+    
     pip install --upgrade pip
-    pip install tox notebook -e '.[notebook-dependencies]'
-    jupyter notebook
+    cd circuit-knitting-toolbox
+
+The next step is to install CKT to the virtual environment. If you plan on running the tutorials, install the
+notebook dependencies in order to run all the visualizations in the notebooks.
+If you plan on developing in the repo, you may want to install the dev dependencies.
```
Comment (garrison) [inline INSTALL.rst:98]:
```suggestion
If you plan on developing in the repository, you may want to install the ``dev`` dependencies.
```

File: INSTALL.rst [hunk 10]
```diff
@@ -1,51 +1,116 @@
...
+
+Users intending to use the entanglement forging tool should install the ``pyscf`` option.
```
Comment (garrison) [inline INSTALL.rst:100]:
```suggestion
Users intending to use the entanglement forging tool should install the ``pyscf`` optional dependency.
```

File: INSTALL.rst [hunk 11]
```diff
@@ -1,51 +1,116 @@
...
 
--  Navigate to the notebooks in the ``docs/tutorials/`` directory to run the
-   tutorials.
+Users intending to use the automatic cut finding functionality in the CutQC package should install the ``cplex`` option.
```
Comment (garrison) [inline INSTALL.rst:102]:
```suggestion
Users intending to use the automatic cut finding functionality in the CutQC package should install the ``cplex`` optional dependency.
```

File: INSTALL.rst [hunk 12]
```diff
@@ -1,51 +1,116 @@
...
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Upgrade pip and install the CKT package.
+
+.. code:: sh
+
+    pip install --upgrade pip
+    pip install circuit-knitting-toolbox
+
+Users intending to use the entanglement forging tool should install the ``pyscf`` option.
```
Comment (garrison) [inline INSTALL.rst:65]:
```suggestion
Users intending to use the entanglement forging tool should install the ``pyscf`` optional dependency.
```

File: INSTALL.rst [hunk 13]
```diff
@@ -1,51 +1,123 @@
...
+    If a user wishes to use the circuit cutting toolbox to
+    automatically find optimized wire cuts for a circuit too large for
+    the free version of CPLEX, they should acquire a license and install
+    the `full
+    version <https://www.ibm.com/products/ilog-cplex-optimization-studio>`__.
+
+Pre-Installation
+^^^^^^^^^^^^^^^^
+
+Users who wish to install locally (using either :ref:`Option 1` or :ref:`Option 2`) may
```
Comment (garrison) [inline INSTALL.rst:29]:
```suggestion
Users who wish to install locally (using either :ref:`Option 1` or :ref:`Option 2`) are encouraged to
```

File: INSTALL.rst [hunk 14]
```diff
@@ -1,51 +1,123 @@
...
+    automatically find optimized wire cuts for a circuit too large for
+    the free version of CPLEX, they should acquire a license and install
+    the `full
+    version <https://www.ibm.com/products/ilog-cplex-optimization-studio>`__.
+
+Pre-Installation
+^^^^^^^^^^^^^^^^
+
+Users who wish to install locally (using either :ref:`Option 1` or :ref:`Option 2`) may 
+follow a few set of common instructions to prepare a Python environment for
```
Comment (garrison) [inline INSTALL.rst:30]:
```suggestion
follow a brief set of common instructions to prepare a Python environment for
```

**Review by garrison — APPROVED** [review]
Thanks!  With this, we can celebrate by closing the 0.2.0 milestone. :tada:
